_aicspylibczi/ImageFactory.h

```diff
@@ -46,7 +46,7 @@ class ImageFactory
 
   static size_t sizeOfPixelType(PixelType pixel_type_);
 
-  static size_t numberOfChannels(PixelType pixel_type_);
+  static size_t numberOfSamples(PixelType pixel_type_);
```

Comment:
Maybe I am missing something but why is this called `size_t` and then `numberOfSamples`? `size_t` as in timepoints?

I 100% bet I am misunderstanding C code so apologies in advance.

Reply:
`size_t` is type like `int` used to represent the size or length of things, especially in the standard library. 
I could have used an unsigned int but because this is value is used in byte offset calculations it made sense to me at least to use size_t.

Reply:
in C++ land there are a whole bunch of integer types with `_t`, you can take it to mean "size_type".  There's uint8_t, uint32_t etc.

Reply:
and size_t here is the return value type of the function.  In c/c++ it's `returnType functionName(argType argName, argType argName, ...)`

Reply:
Got it. Thanks.